### ISSUE TITLE
Unexpected situation arose thrown when module transformer is enabled.

### ISSUE THREAD

**Issue by swernerx** [rollup/rollup-plugin-babel#220]
There seems to be a major issue right now when accidiently enabled the module transformer. 

It just throws with this error:

```
An unexpected situation arose. Please raise an issue at https://github.com/rollup/rollup-plugin-babel/issues. Thanks!
```

which is neither helpful or correct as it seems.

In this case it seems that Babel Plugin Rollup is unable to match the generated code to one of the snippets. The code throwing this error looks like this:

```js
var inputCode = 'class Foo extends Bar {};\nexport default Foo;';
var check = core.transform( inputCode, options ).code;

if ( ~check.indexOf('class ') ) {
	options.plugins = (options.plugins || []).concat( fallbackClassTransform );
	check = core.transform( inputCode, options ).code;
}

if ( ~check.indexOf( 'import _inherits' ) ) { helpers = RUNTIME; }
else if ( ~check.indexOf( 'function _inherits' ) ) { helpers = INLINE; }
else if ( ~check.indexOf( 'babelHelpers' ) ) { helpers = EXTERNAL; }
else {
	throw new Error( 'An unexpected situation arose. Please raise an issue at https://github.com/rollup/rollup-plugin-babel/issues. Thanks!' );
}
```

I understand that the code has to check something here to figure out wether to target helpers, the runtime, etc.

The code being generated when modules are transformed looks like this:

```js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/builtin/interopRequireDefault");

exports.__esModule = true;
exports.default = void 0;

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/builtin/inherits"));

class Foo extends Bar {}

var _default = Foo;
exports.default = _default;
```

I figure we could improve the situation by adding alternative matchers to the if-else construct:

```js
if ( ~check.indexOf( 'import _inherits' ) || ~check.indexOf( 'helpers/builtin/inherits' ) ) { helpers = RUNTIME; }
```

This modification is for Babel v7 with runtime helpers enabled. I am not sure whether there is something we could change for the other options, too.

After adding this the real error from Rollup is showing up that the module transformer is the real issue:

```
Error: It looks like your Babel configuration specifies a module transformer. Please disable it. 
```

**Comment by Andarist** [comment]
Would love to accept a PR to improve the current situation 😉 Although the additional check could be improved (I think). Are u sure that what u have pasted is the output? `_inherits2` doesnt seem to be used.

**Comment by swernerx** [comment]
I'll have a look!

**Comment by swernerx** [comment]
Playing with your test suite... adding my preset to one of the class tests I see the following output code being generated:

```js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/builtin/interopRequireDefault");

exports.__esModule = true;
exports.default = void 0;

var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/builtin/inheritsLoose")),
    Foo =
/*#__PURE__*/
function (_Bar) {
  (0, _inheritsLoose2.default)(Foo, _Bar);

  function Foo() {
    return _Bar.apply(this, arguments) || this;
  }

  return Foo;
}(Bar);

var _default = Foo;
exports.default = _default;
```

WIP

**Comment by swernerx** [comment]
Seems like I found the smallest .babelrc combination which crashes:

```js
{
  "presets": [ "@babel/env" ],
  "plugins": [ "@babel/external-helpers", "@babel/transform-runtime" ]
}
```

**Comment by swernerx** [comment]
PR is ready. Please check whether this is okay. Thanks!

**Comment by Andarist** [comment]
Fixed by #225
